docs/specs/kemsphinx.rst

```diff
@@ -30,7 +32,69 @@ has two functions:
 Therefore we must embed these KEM ciphertexts in the KEMSphinx header,
 one KEM ciphertext per mix hop.
 
-2. KEMSphinx Header Design
+2. Post Quantum Hybrid KEM
+==========================
+
+Special care must be taken in order correctly compose a hybrid post
+quantum KEM that is IND-CCA2 robust.
+
+The hybrid post quantum KEMs found in Cloudflare's circl library are
+suitable to be used with Noise or TLS but not with KEM Sphinx because
+they are not IND-CCA2 robust. Noise and TLS achieve IND-CCA2 security
+by mixing in the public keys and ciphertexts into the hash object and
+therefore do not require an IND-CCA2 KEM.
+
+Firstly, our post quantum KEM is IND-CCA2 however we must specifically take care
+to make our NIKE to KEM adapter have semantic security. Secondly, we must make a
+security preserving KEM combiner.
+
+2.1 NIKE to KEM adapter
+-----------------------
+
+We easily achieve our IND-CCA2 security by means of hashing
+together the DH shared secret along with both of the public keys:
+
+```
+func ENCAPSULATE(their_pub_key publickey) ([]byte, []byte) {
+        my_priv_key, my_pub_Key = GEN_KEYPAIR(RNG)
+        ss = DH(my_priv_key, their_pub_key)
+        ss2 = H(ss || my_pubkey || their_pubkey)
+	return my_pub_key, ss2
+}
+
+func DECAPSULATE(my_priv_key, their_pub_key) []byte {
+        s = DH(my_priv_key, their_pub_key)
+	shared_key = H(ss || my_pub_key || their_pub_key)
+	return shared_key
+}
+```
+
+
+2.2 KEM Combiner
+----------------
+
+The KEM Combiners paper [KEMCOMB]_ makes the observation that if a KEM combiner
+is not security preserving then the resulting hybrid KEM will not have
+IND-CCA2 security if one of the composing KEMs does not have IND-CCA2
+security. Likewise the paper points out that when using a security
+preserving KEM combiner, if only one of the composing KEMs has
+IND-CCA2 security then the resulting hybrid KEM will have IND-CCA2
+security.
+
+Our KEM combiner uses the split PRF design from the paper when combining
+two KEM shared secrets together we use a hash function to also mix
+in the values of both KEM ciphertexts:
+
```

Comment:
- I think here we should mention what `cct1`, `cct2`, `ss1` and `ss2` are, ie they're the values from `ENCAPSULATE()` of the underlying KEMs `1` and `2`, and `cct{1,2}` are the ephemeral envelope public keys ?

Reply:
Yes, exactly. cct1 and cct2 are KEM ciphertexts. Although one of them is from the NIKE to KEM adapter.

Reply:
Cool, we should put that in the spec so implementers can do it clean-room without having to go scouring the implementation to figure out what `cct` and `ss` are supposed to be.

Reply:
Sorry, I was reading the old commit. New text in `2.2` does explain it, thanks!